pinot-core/src/main/java/org/apache/pinot/core/common/ColumnValueReader.java

```diff
@@ -0,0 +1,248 @@
+   * <p>NOTE: Dictionary id is handled as INT type.
+   */
+  public int getIntValue(int docId) {
+    return _forwardIndexReader.getInt(docId, _readerContext);
+  }
+
+  /**
+   * Returns the LONG type single-value at the given document id.
+   */
+  public long getLongValue(int docId) {
+    return _forwardIndexReader.getLong(docId, _readerContext);
+  }
+
+  /**
+   * Returns the FLOAT type single-value at the given document id.
+   */
+  public float getFloatValue(int docId) {
+    return _forwardIndexReader.getFloat(docId, _readerContext);
+  }
+
+  /**
+   * Returns the DOUBLE type single-value at the given document id.
+   */
+  public double getDoubleValue(int docId) {
+    return _forwardIndexReader.getDouble(docId, _readerContext);
+  }
+
+  /**
+   * Returns the STRING type single-value at the given document id.
+   */
+  public String getStringValue(int docId) {
+    return _forwardIndexReader.getString(docId, _readerContext);
+  }
+
+  /**
+   * Returns the BYTES type single-value at the given document id.
+   */
+  public byte[] getBytesValue(int docId) {
+    return _forwardIndexReader.getBytes(docId, _readerContext);
+  }
+
+  /**
+   * Reads the INT type multi-value at the given document id into the value buffer and returns the number of values in
+   * the multi-value entry.
+   * <p>The passed in value buffer should be large enough to hold all the values of a multi-value entry.
+   * <p>NOTE: Dictionary id is handled as INT type.
+   */
+  public int getIntValues(int docId, int[] valueBuffer) {
+    return _forwardIndexReader.getIntArray(docId, valueBuffer, _readerContext);
+  }
+
+  // TODO: Support raw index for multi-value columns
+
+  /**
+   * NOTE: The following batch value read APIs support data type conversions for numeric types. Caller can call any
+   *       API regardless of the value type.
+   * TODO: Consider letting the caller handle the data type conversion because for different use cases, we might need to
+   *       convert data type differently.
+   */
+
+  /**
+   * Batch reads the INT type single-values at the given document ids of the given length into the value buffer.
+   * <p>The passed in value buffer size should be larger than or equal to the length.
+   * <p>NOTE: Dictionary id is handled as INT type.
+   */
+  public void getIntValues(int[] docIds, int length, int[] valueBuffer) {
+    if (_valueType == DataType.INT) {
+      _forwardIndexReader.readValues(docIds, length, valueBuffer, _readerContext);
+    } else {
+      throw new IllegalStateException(String.format("Cannot read %s as INT", _valueType));
+    }
+  }
+
+  /**
+   * Batch reads the LONG type single-values at the given document ids of the given length into the value buffer.
+   * <p>The passed in value buffer size should be larger than or equal to the length.
+   */
+  public void getLongValues(int[] docIds, int length, long[] valueBuffer) {
+    switch (_valueType) {
+      case INT:
```

Comment:
Does it make sense to read strings as numbers (of course, assuming that these are numbers stored as strings)? Or that use case does not make sense?

Reply:
We can try to cast all types, but could be costly (there is a TODO on line 126).
Again, this PR only involves clean up and refactor without any functionality change. We can add that in a separate PR if required.